Turborepo support (#557)
railwayapp/nixpacks: docs/pages/docs/providers/node.md

The build script found in `package.json` if it exists or if its an NX Monorepo `(npm|pnpm|yarn|bun) run build <NxAppName> --configuration=production`.

> Or, if it's a Turborepo monorepo (detected if `turbo.json` exists), the `build` pipeline will be called, if it exists; otherwise, the `build` script of the package referenced by `NIXPACKS_TURBO_APP_NAME` will be called, if the variable is provided; otherwise, it will fall back to the build script found in `package.json`.

Reviewer: ```suggestion
Or, if it's a Turborepo monorepo (detected if `turbo.json` exists), the `build` pipeline will be called (if it exists). Otherwise, the `build` script of the `package.json` referenced by `NIXPACKS_TURBO_APP_NAME` will be called, if `NIXPACKS_TURBO_APP_NAME` is provided and valid. Otherwise, it will fall back to the build script found in `package.json` at the monorepos root.
```

Author: It doesn't actually check the validity of `NIXPACKS_TURBO_APP_NAME`. Should it? I don't think that the NX code does.

Reviewer: It should. I'll implement support for Nx and validating the name in another PR.

  - If the app is a NextJS project: `npm run start`
  - If `targets.build.options.main` exists in the apps `Project.json`: `node <outputPath>/<mainFileName>.js` (e.g `node dist/apps/my-app/main.js`)
  - Fallback: `node <outputPath>/index.js` (e.g `node dist/apps/my-app/index.js`)

> - If it's a Turborepo

Reviewer: ```suggestion
- If Turborepo is detected
```
